diff --git a/src/droidy/injector.vala b/src/droidy/injector.vala
index 263f608c..2bfd181d 100644
--- a/src/droidy/injector.vala
+++ b/src/droidy/injector.vala
@@ -81,7 +81,7 @@ namespace Frida.Droidy.Injector {
 			string so_path_app = "/data/data/" + package + "/gadget.so";
 			string config_path_shared = "/data/local/tmp/frida-gadget-" + instance_id + ".config";
 			string config_path_app = "/data/data/" + package + "/gadget.config";
-			string unix_socket_path = "frida:" + package;
+			string unix_socket_path = "banana:" + package;
 
 			bool waiting = false;
 			uint target_pid = 0;
diff --git a/src/linux/frida-helper-backend.vala b/src/linux/frida-helper-backend.vala
index d88405b6..30fecefb 100644
--- a/src/linux/frida-helper-backend.vala
+++ b/src/linux/frida-helper-backend.vala
@@ -1207,9 +1207,9 @@ namespace Frida {
 			return (void *) (remote_module.base_address + offset);
 		}
 
-		private static string make_fallback_address () {
-			return "/frida-" + Uuid.string_random ();
-		}
+	private static string make_fallback_address () {
+		return "/banana-" + Uuid.string_random ();
+	}
 
 		private Future<RemoteAgent> establish_connection (LoaderLaunch launch, InjectSpec spec, BootstrapResult bres,
 				UnixConnection? agent_ctrl, string fallback_address, Cancellable? cancellable) throws Error, IOError {
diff --git a/src/linux/frida-helper-process.vala b/src/linux/frida-helper-process.vala
index 856a7149..e023af12 100644
--- a/src/linux/frida-helper-process.vala
+++ b/src/linux/frida-helper-process.vala
@@ -321,7 +321,7 @@ namespace Frida {
 			TimeoutSource? timeout_source = null;
 
 			try {
-				string socket_path = "/frida-" + Uuid.string_random ();
+				string socket_path = "/banana-" + Uuid.string_random ();
 				string socket_address = "unix:abstract=" + socket_path;
 
 				service = new SocketService ();
